src/main.rs

```diff
@@ -1,171 +1,17 @@
-                let node = notification.get_node();
-                println!("NodeAdded: {:?}", node);
-                state.add_node(node);
-            },
-            NotificationType::Type_NodeRemoved => {
-                let mut state = self.state.lock().unwrap();
-                let node = notification.get_node();
-                println!("NodeRemoved: {:?}", node);
-                state.remove_node(node);
-            },
-            NotificationType::Type_NodeEvent => {
-                println!("NodeEvent");
-            },
-            NotificationType::Type_ValueAdded => {
-                let mut state = self.state.lock().unwrap();
-                let value_id = notification.get_value_id();
-                println!("ValueAdded: {:?}", value_id);
-                state.add_value_id(value_id);
-            },
-            NotificationType::Type_ValueChanged => {
-                let mut state = self.state.lock().unwrap();
-                let value_id = notification.get_value_id();
-                println!("ValueChanged: {:?}", value_id);
-                state.add_value_id(value_id);
-                // TODO: Tell somebody that the value changed
-            },
-            NotificationType::Type_ValueRemoved => {
-                let mut state = self.state.lock().unwrap();
-                let value_id = notification.get_value_id();
-                println!("ValueRemoved: {:?}", value_id);
-                state.remove_value_id(value_id);
-            },
-            _ => {
-                //println!("Unknown notification: {:?}", notification);
-            }
-        }
-    }
-}
+use std::io;
+use std::io::Write;
 
 fn main() {
-    let mut options = options::Options::create("./config/", "", "--SaveConfiguration true --DumpTriggerLevel 0 --ConsoleOutput false").unwrap();
-
-    // TODO: The NetworkKey should really be derived from something unique
-    //       about the foxbox that we're running on. This particular set of
-    //       values happens to be the default that domoticz uses.
-    options::Options::add_option_string(&mut options, "NetworkKey", "0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10", false).unwrap();
 
-    let mut manager = manager::Manager::create(options).unwrap();
-    let program = Program::new();
-
-    manager.add_watcher(program.clone()).unwrap();
-
-    {
-        let arg_device: Option<String> = std::env::args()
-            .skip(1).last(); // last but not first
-
-        let device = match arg_device {
-            Some(ref x) => x as &str,
-            None => get_default_device().expect("No device found.")
-        };
-
-        println!("found device {}", device);
-
-        match device {
-            "usb" => manager.add_usb_driver(),
-            _ => manager.add_driver(&device)
-        }.unwrap()
-    }
+    let options = InitOptions {
+        device: std::env::args().skip(1).last() // last but not first
```

Comment:
not completely sure about this yet; maybe we want an Option. Maybe we want to be able to add this later with a function. So this is like it for now.